### PR TITLE
fix: clarify the update steps in canary with traffic routing

### LINKED ISSUE #1405: Istio Subset- Level traffic splitting cannot be used for Rolling Update
## Summary 
Update process does not Scale down the old RS, Its renewal process is more like blue-green.
It waits for all new ReplicaSet pods  to started and scale down the old ReplicaSet Pods to 0.
```yaml
strategy:
  canary:
    # maxSurge: "25%"
    # maxUnavailable: "25%"
    scaleDownDelaySeconds: 30
    trafficRouting:
      istio:
        virtualService:
          name: helloworld-vsvc # required
          routes:
            - primary # optional if there is a single route in VirtualService, required otherwise
        destinationRule:
          name: helloworld-destrule # required
          canarySubsetName: canary # required
          stableSubsetName: stable # required
    steps:
      - setWeight: 20
      - pause: {duration: 20s}
      - setWeight: 40
      - pause: {duration: 20s}
      - setWeight: 60
      - pause: {duration: 20s}
      - setWeight: 80
      - pause: {duration: 20s}
```

## Diagnostics

What version of Argo Rollouts are you running?
1.0.4

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

### PR DESCRIPTION
**Pull request by huikang**
Signed-off-by: Hui Kang <hui.kang@salesforce.com>

close #1405 

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

**Comment by huikang** [comment]
@harikrongali , could you take a look at this one? Thanks.

**Comment by harikrongali** [comment]
LGTM

File: docs/getting-started/istio/index.md [hunk 1]
```diff
@@ -149,3 +149,8 @@ spec:
 
 As the Rollout progresses through steps, the HTTP route destination weights will be adjusted to
 match the current setWeight of the steps.
+
+
+!!! tip
+  Unlike the ReplicaSet based weighted canary, a service mesh/ingress based canary leaves the
+  stable as 100% scaled until the rollout completes.
```
Comment (jessesuen) [inline docs/getting-started/istio/index.md:156]: NOTE: this is actually going to change with PR https://github.com/argoproj/argo-rollouts/pull/1382, which will give the user the option to scale down the stable ReplicaSet, as the canary scales up. 

So this note should get updated when v1.1 is released. Do you still want to add this note?

Comment (huikang) [reply docs/getting-started/istio/index.md:156]: You are right, @jessesuen . Thanks.

**Pull request closed without merging**